profiler/src/main/java/com/splunk/opentelemetry/profiler/RecordingSequencer.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.splunk.opentelemetry.profiler.util.HelpfulExecutors;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Responsible for periodically generating a sequence of JFR recording files. Prior to starting a
+ * recording, it consults with a RecordingEscapeHatch to make sure that it is safe/relevant to do.
+ */
+public class RecordingSequencer {
+  private static final Logger logger = LoggerFactory.getLogger(RecordingSequencer.class.getName());
+  // The overlap factor causes recordings to be shorter than the requested duration.
+  // This forces overlap between recordings and ensures data is continuous when deduplicated.
+  public static final double OVERLAP_FACTOR = 0.8;
+
+  private final ScheduledExecutorService executor =
+      HelpfulExecutors.newSingleThreadedScheduledExecutor("JFR Recording Sequencer");
+  private final Duration recordingDuration;
+  private final RecordingEscapeHatch recordingEscapeHatch;
+  private final JfrRecorder recorder;
+
+  public RecordingSequencer(Builder builder) {
+    this.recordingDuration = builder.recordingDuration;
+    this.recordingEscapeHatch = builder.recordingEscapeHatch;
+    this.recorder = builder.recorder;
+  }
+
+  public void start() {
+    int period = (int) (recordingDuration.toMillis() * OVERLAP_FACTOR);
+    executor.scheduleAtFixedRate(this::handleInterval, 0, period, TimeUnit.MILLISECONDS);
+  }
+
+  @VisibleForTesting
+  void handleInterval() {
+    if (!recordingEscapeHatch.jfrCanContinue()) {
+      logger.warn("JFR recordings cannot proceed.");
```

Comment:
will this end up logging every 20s? Should we keep a reference to the Future from the schedule and cancel it when we see this?

Reply:
It will. The hope was to make it very obvious that something bad has happened and that it's broken.  Think disk filling up or completely backlogged (too many files bunching up).  It should be common, but we wanted to guard against it.

I don't want to cancel the schedule, because we want it to recover if things clear up.

If we hate that repeated logging, I supposed we could keep additional state and only log a max `n` times or something.

Reply:
ah, ok. you expect it to be a transient issue (although disk filling up rarely is!). That makes sense.

Reply:
It was hard to tell from this code whether the escape hatch might be something that the user pulls during runtime to turn off JFR.

Reply:
Nah, the request was for it to pause itself when things are bad.